sdk/core/core-rest-pipeline/src/policies/throttlingRetryPolicy.ts

```diff
@@ -22,21 +27,23 @@ export function throttlingRetryPolicy(): PipelinePolicy {
   return {
     name: throttlingRetryPolicyName,
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      const response = await next(request);
-      if (response.status !== 429) {
-        return response;
-      }
+      let response = await next(request);
 
-      const retryAfterHeader = response.headers.get("Retry-After");
-
-      if (retryAfterHeader) {
+      for (let count = 0; count <= DEFAULT_CLIENT_MAX_RETRY_COUNT; count++) {
```

Comment:
won't `<=` cause it to execute too many times? `0, 1, 2, 3` is four retries

Reply:
In this case, 0 is the first request, 1, 2, 3 are the retries 🤔 or should 0 be counted as a retry?

Reply:
hngggh okay you're right

Reply:
wait no! the first request was line 30 no?

Reply:
That is true! Alright, alright! Tests will fix this for sure though :)